.github/workflows/check-commits.yml

```diff
@@ -61,3 +62,26 @@ jobs:
               repo: context.repo.repo,
               body: output
             })
+  # check that only released Gradle versions are used in the wrapper
+  check_wrapper:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check used wrapper
+        run: |
+          set -eu
+          for commit in $(git rev-list ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}); do
+              git checkout $commit
```

Comment:
💅 maybe add `--quiet` and `--detach` to remove lots of output that may distract from the actual relevant outputs